include/openPMD/Series.hpp

```diff
@@ -282,5 +348,139 @@ class Series : public Attributable
     std::shared_ptr< std::string > m_filenamePrefix;
     std::shared_ptr< std::string > m_filenamePostfix;
     std::shared_ptr< int > m_filenamePadding;
+
+    std::shared_ptr< auxiliary::Option< WriteIterations > > m_writeIterations =
+        std::make_shared< auxiliary::Option< WriteIterations > >();
 }; // Series
+
+/**
+ * @brief Subclass of Iteration that knows its own index withing the containing
+ *        Series.
+ */
+class IndexedIteration : public Iteration
+{
+    friend class SeriesIterator;
+
+public:
+    using iterations_t = decltype( Series::iterations );
+    using index_t = iterations_t::key_type;
+    index_t const iterationIndex;
+
+private:
+    template< typename Iteration_t >
+    IndexedIteration( Iteration_t && it, index_t index )
+        : Iteration( std::forward< Iteration_t >( it ) )
+        , iterationIndex( index )
+    {
+    }
+};
+
+class SeriesIterator
+{
+    using iteration_index_t = IndexedIteration::index_t;
+
+    using maybe_series_t = auxiliary::Option< Series * >;
+
+    maybe_series_t m_series;
+    iteration_index_t m_currentIteration = 0;
+
+    // construct the end() iterator
+    SeriesIterator();
+
+public:
+    SeriesIterator( Series * );
+
+    SeriesIterator &
+    operator++();
+
+    IndexedIteration
+    operator*();
+
+    bool
+    operator==( SeriesIterator const & other ) const;
+
+    bool
+    operator!=( SeriesIterator const & other ) const;
+
+    static SeriesIterator
+    end();
+};
+
+/**
+ * @brief Reading side of the streaming API. Create instance via
```

Comment:
See details below, please.

```suggestion
/** Reading side of the streaming API.
 *
 * Create instance via
```